snuba/query/snql/__init__.py

```diff
@@ -29,19 +35,23 @@
     visit_parameters_list,
     visit_quoted_literal,
 )
-from typing import Sequence
-
 
 snql_grammar = Grammar(
     fr"""
-    query_exp             = match_clause where_clause? collect_clause? group_by_clause? having_clause? order_by_clause?
+    query_exp             = match_clause where_clause collect_clause group_by_clause? having_clause? order_by_clause?
 
     match_clause          = space* "MATCH" space* open_paren clause close_paren space*
-    where_clause          = space* "WHERE" clause space*
+    where_clause          = space* "WHERE" or_expression space*
     collect_clause        = space* "COLLECT" collect_list space*
     group_by_clause       = space* "BY" group_list space*
+    having_clause         = space* "HAVING" or_expression space*
     order_by_clause       = space* "ORDER BY" order_list space*
 
+    condition             = low_pri_arithmetic condition_op (column_name / numeric_literal)
+    condition_op          = "=" / "!=" / ">" / ">=" / "<" / "<="
```

Comment:
Should "IN" be on this list as well?

Reply:
Currently, no, I'm handling the other conditional operators first. The next step is to deal with parenthetical expressions (those that indicate priority), and if time permits, I would deal with the other operators like "IN".